Nodes/HTTPSyncNode.cpp

```diff
@@ -0,0 +1,284 @@
+
+    try {
+        setShipID( m_dbHandler->retrieveCell("server", "1", "boat_id") );
+        setShipPWD( m_dbHandler->retrieveCell("server", "1", "boat_pwd") );
+        setServerURL( m_dbHandler->retrieveCell("server", "1", "srv_addr") );
+
+        m_initialised = true;
+
+    } catch (const char * error) {
+        Logger::error("%s Failed to get server configuration Error: %s", __PRETTY_FUNCTION__, error);
+        //Kill thread if setup fails
+        std::terminate();
+    }
+    Logger::info("HTTPSyncNode init() successful");
+
+    return m_initialised;
+
+}
+
+void HTTPSyncNode::start(){
+
+    if (m_initialised)
+    {
+        runThread(HTTPSyncThread);
+    }
+    else
+    {
+        Logger::error("%s Cannot start HTTPSYNC thread as the node was not correctly initialised!", __PRETTY_FUNCTION__);
+    }
+
+}
+
+void HTTPSyncNode::processMessage(const Message* msgPtr)
+{
+    MessageType msgType = msgPtr->messageType();
+
+    switch(msgType)
+    {
+        case MessageType::LocalWaypointChange:
+            pushWaypoints();
+            break;
+        case MessageType::LocalConfigChange:
+            pushConfigs();
+            break;
+        default:
+            break;
+        //Potentially: on new log data. Currently sending constantly in thread
+    }
+
+}
+
+void HTTPSyncNode::HTTPSyncThread(void* nodePtr){
+
+    HTTPSyncNode* node = (HTTPSyncNode*)(nodePtr);
+
+    Logger::info("HTTPSync thread has started");
+
+    while(true)
+    {
+        node->getConfigsFromServer();
+        node->getWaypointsFromServer();
+        node->pushDatalogs();
+
+        std::this_thread::sleep_for(std::chrono::milliseconds(node->m_delay));
+    }
+
+    Logger::info("HTTPSync thread has exited");
+}
+
+void HTTPSyncNode::pushDatalogs() {
+    std::string response = "";
+    if(performCURLCall(m_dbHandler->getLogs(m_pushOnlyLatestLogs), "pushAllLogs", response))
+    {
+         //remove logs after push
+        if(m_removeLogs) {
+            m_dbHandler->clearLogs();
+        }
+
+        Logger::info("Logs pushed to server");
+    }
+    else if(!m_reportedConnectError)
+    {
+        Logger::warning("%s Could not push logs to server:", __PRETTY_FUNCTION__);
+    }
+}
+
+void HTTPSyncNode::pushWaypoints() 
+{
+	std::string waypointsData = m_dbHandler->getWaypoints();
+	if (waypointsData.size() > 0)
+	{
+		std::string response;
+		if(performCURLCall(waypointsData,"pushWaypoints", response))
+		{
+			Logger::info("Waypoints pushed to server");
+		}
+		else if(!m_reportedConnectError)
+		{
+			Logger::warning("%s Failed to push waypoints to server", __PRETTY_FUNCTION__);
+		}
+	}
+}
+
+void HTTPSyncNode::pushConfigs() {
+    std::string response;
+
+	if(performCURLCall(m_dbHandler->getConfigs(), "pushConfigs", response))
+	{
+		Logger::info("Configs pushed to server");
+	}
+	else if(!m_reportedConnectError)
+	{
+		Logger::warning("%s Error: ", __PRETTY_FUNCTION__);
+	}
+}
+
+
+void HTTPSyncNode::setShipID(std::string ID) {
```

Comment:
Are these functions necessary, could they not just be set in init or the constructor?